### DISCUSSION TITLE
Support for Deno

### DISCUSSION THREAD

**Discussion by PaulKiddle** [webdriverio/webdriverio#6741] [Q&A] [answered]
Hi there, I've seen a couple of references to Deno support for this library, is there any documentation on how to use webdriver in a Deno environment?

**Comment by christian-bromann** [comment, marked as answer]
Hey @PaulKiddle , this is an effort we were already pursuing. However we then prioritised the transformation to TypeScript. If you want to help us please get involved, see: https://github.com/webdriverio/webdriverio/projects/17

**Reply by erwinheitzman** [reply]
Deno runs TypeScript natively so I believe that when running TypeScript, it might already work out of the box. Not sure if anything needs to change when running Javascript

**Reply by christian-bromann** [reply]
I believe Deno doesn't support the CommonJS module system which we use in this codebase.

**Comment by jesusfj710** [comment]
Whats the status of this? I can not find any information about the progress in Deno in WDIO D:

**Reply by christian-bromann** [reply]
I haven't revisited this since then, now that WebdriverIO is run in ESM and Deno supports NPM modules it might be already possible. Have you tried?

**Reply by jesusfj710** [reply]
Well I took a look at it but I don't see how to do a very basic point. Deno lets you import the packages and you can run a `deno run hello_world.ts` which uses any library imported from NPM and no problem.

What I don't see clearly is how to run the equivalent of `yarn wdio wdio.config.ts`, since in Deno you don't really install WDIO in the same way and you can't do a `deno run wdio wdio.config.ts` or `deno run wdio.config.ts`.

**Reply by christian-bromann** [reply]
How do other testing frameworks would work in Deno?

**Reply by jesusfj710** [reply]
To be honest, I tried to find some information about and I didnt get anything.
